src/dotnet.ts

```diff
@@ -0,0 +1,105 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as vscode from 'vscode';
+import * as which from 'which';
+import * as cp from 'child_process';
+import * as semver from 'semver';
+import * as path from 'path';
+
+export type DotNetInfo = {path: string, version: string};
+export type DotNetCommandResult = {code: number, msg: string};
+
+let dotnetInfo: DotNetInfo | undefined = undefined;
+
+function promptToInstallDotNetCoreSDK(msg: string): void {
+    let installItem = 'Install .NET Core SDK...';
+    vscode.window
+        .showErrorMessage(msg, installItem)
+        .then(
+            (item) => {
+                if (item === installItem) {
+                    vscode.env.openExternal(vscode.Uri.parse('https://dotnet.microsoft.com/download'));
+                }
+            }
+        );
+}
+
+/**
+ * Returns the path to the .NET Core SDK, or prompts the user to install
+ * if the SDK is not found.
+ */
+export function findDotNetSdk(): Promise<DotNetInfo> {
+    return new Promise((resolve, reject) => {
+
+        if (dotnetInfo === undefined) {
+            try {
+                let path = which.sync('dotnet');
+                cp.exec(
+                    `"${path}" --version`,
```

Comment:
I don't think cp package takes care of any escaping. But we are getting the path using which utility which would look in PATH environment variable and should return a platform specific path. What else do you think we should do to make it robust?

Reply:
I think double-quoting for path itself should do (at least Windows and Linux, not sure about MacOS), but whenever you pass any dynamically created string as an argument - you need to make sure to quote it, if it has spaces and while doing so escape double-quotes themselves.
Are you sure ```cp.spawn``` doesn't do it? I'm certain ```cp.exec``` won't do anything, because it expects the whole command to be passed as a single string, but whenever you pass executable and arguments separately and they concatenate themselves - I'd expect them to do due diligence to quote and escape everything...

Is there a reason why we can't use ```spawn``` here?

Reply:
Generally exec is recommended when the output is limited and not a stream as exec returns buffer. There is a property in spawn windowsVerbatimArguments when set to true it will quote and escape. I will try that.

Reply:
Do you know if it works cross-platform? I don't llike ```windows``` in the name of the property... Also, if there is no dynamic strings/user input in the arguments (i.e. they are static strings and you know what they are), then it might be okay to just pass as-is. The problem is typically with spaces - as long as there is user provided string - you need to check for spaces and quote if they are there...

Reply:
Found this: https://github.com/nodejs/node/issues/6803
Yes that argument is simply ignored on linux. 
I think it is a good idea to quote it even though it is not user provided string. This will make sure spaces are escaped.